pySDC/tests/test_projects/test_DAE/test_problems.py

```diff
@@ -325,3 +325,377 @@ def test_synchgen_infinite_bus_main():
+
+    P = controller.MS[0].levels[0].prob
+    uinit = P.u_exact(t0)
+    uex = P.u_exact(Tend)
+
+    uend, stats = controller.run(u0=uinit, t0=t0, Tend=Tend)
+
+    err = abs(uex[0] - uend[0])
+    assert err < err_tol[M], f"ERROR for M={M}: Error is too large! Expected {err_tol[M]}, got {err}"
+
+    switches = get_sorted(stats, type='switch', sortby='time', recomputed=False)
+    assert len(switches) >= 1, 'ERROR for M={M}: No events found!'
+    t_switches = [item[1] for item in switches]
+    t_switch = t_switches[-1]
+
+    t_switch_exact = P.t_switch_exact
+    event_err = abs(t_switch_exact - t_switch)
+    assert (
+        event_err < event_err_tol[M]
+    ), f"ERROR for M={M}: Event error is too large! Expected {event_err_tol[M]}, got {event_err}"
+
+
+@pytest.mark.base
+def test_WSCC9_SDC_detection():
+    """
+    Test for one SDC run with event detection if the found event is close to the exact value and if the global error
+    can be reduced.
+    """
+
+    from pySDC.helpers.stats_helper import get_sorted
+    from pySDC.projects.DAE.problems.WSCC9BusSystem import WSCC9BusSystem
+    from pySDC.projects.DAE.sweepers.fully_implicit_DAE import fully_implicit_DAE
+    from pySDC.implementations.controller_classes.controller_nonMPI import controller_nonMPI
+    from pySDC.projects.PinTSimE.switch_estimator import SwitchEstimator
+    from pySDC.implementations.convergence_controller_classes.basic_restarting import BasicRestartingNonMPI
+    from pySDC.implementations.hooks.log_solution import LogSolution
+
+    ref = {
+        'Eqp': [1.02565963, 0.87077674, 0.75244422],
+        'Si1d': [1.08501824, 0.8659459, 0.59335005],
+        'Edp': [-9.89321652e-26, 3.55754231e-01, 5.66358724e-01],
+        'Si2q': [0.03994074, -0.380399, -0.67227838],
+        'Delta': [-2.13428304, 10.32368025, -1.48474241],
+        'w': [370.54298062, 398.85092866, 368.59989826],
+        'Efd': [1.33144618, 2.11434102, 2.38996818],
+        'RF': [0.22357495, 0.35186554, 0.36373663],
+        'VR': [1.3316767, 2.48163506, 2.97000777],
+        'TM': [0.98658474, 0.63068939, 1.12527586],
+        'PSV': [1.0, 0.52018862, 1.24497292],
+        'Id': [1.03392984e00, -7.55033973e-36, 1.39602103e00],
+        'Iq': [1.80892723e00, 1.15469164e-30, 6.38447393e-01],
+        'V': [
+            0.97014097,
+            0.94376174,
+            0.86739643,
+            0.9361775,
+            0.88317809,
+            0.92201319,
+            0.83761267,
+            0.85049254,
+            0.85661891,
+        ],
+        'TH': [
+            -2.30672821,
+            9.90481234,
+            -2.45484121,
+            -2.42758466,
+            -2.57057159,
+            -2.4746599,
+            -2.67639373,
+            -2.62752952,
+            -2.5584944,
+        ],
+        't_switch': [0.5937503078440701],
+    }
+
+    level_params = {
+        'restol': 5e-13,
+        'dt': 1 / (2**5),
+    }
+
+    problem_params = {
+        'newton_tol': 1e-10,
```

Comment:
Is the Newton tolerance intentionally larger than the residual tolerance? I understand it's something different than I do, so maybe this makes sense. But the residual tolerance is really very low.

Reply:
I observed that SDC for DAEs converges very slowly, so setting a smaller residual tolerance could mean that you do unnecessary iterations while not having a benefit of reaching a smaller residual tolerance. And setting a Newton tolerance would mean that you have higher computational costs applying your Newton solver. This system is already very expensive to solve. 
But setting the residual tolerance too large you cannot expect a small error. Especially around the discrete event it is necessary to have accuracy.

Reply:
SDC solves a collocation problem to some accuracy inside the Newton iterations, right? Then a residual tolerance smaller or equal to the Newton tolerance does make sense. I am just trying to understand what you are doing.